src/components/TodoItem.tsx

```diff
@@ -0,0 +1,146 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import cn from 'classnames';
+
+import {
+  useContext, useEffect, useRef, useState,
+} from 'react';
+import { Todo } from '../types/Todo';
+import { DispatchContext } from './TodosContext';
+
+type Props = {
+  todo: Todo,
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const { completed, id, title } = todo;
+
+  const [isEditing, setIsEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(title);
+
+  const dispatch = useContext(DispatchContext);
+  const titleRef = useRef<HTMLInputElement>(null);
+
+  const toggleStatus = (e: React.ChangeEvent<HTMLInputElement>) => {
+    dispatch({
+      type: 'updateTodo',
+      payload: {
+        id,
+        title,
+        completed: e.target.checked,
+      },
+    });
+  };
+
+  const handleDestroy = () => {
+    dispatch({
+      type: 'destroy',
+      payload: todo.id,
+    });
+  };
+
+  const handleDoubleClick = () => {
+    setNewTitle(title);
+    setIsEditing(true);
+  };
+
+  const handleKeyUp = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    switch (e.key) {
+      case 'Enter': {
```

Comment:
Enter is a 'magic word'
Consider will be better if you create a enum for magic words and use it everywhere